File: 2016-maximum-difference-between-increasing-elements/2016-maximum-difference-between-increasing-elements.py
class Solution:
    def maximumDifference(self, nums: List[int]) -> int:
        i = 0
        maxDifference = -1
        for j in range(1,len(nums)):
            if nums[j] > nums[i]:
                maxDifference = max(maxDifference, nums[j] - nums[i])
            else:
                i = j
        return maxDifference

File: 2016-maximum-difference-between-increasing-elements/2016-maximum-difference-between-increasing-elements.go
func maximumDifference(nums []int) int {
    i := 0
	maxDifference := -1
	for j := 1 ; j < len(nums) ; j++ {
		if nums[j] > nums[i] {
			if nums[j] - nums[i] > maxDifference {
				maxDifference = nums[j] - nums[i]
			}
		} else {
			i = j
		}
	}
	return maxDifference
}